### PR TITLE
feature: ubuntu 24.04 and node22 for docker image

### PR DESCRIPTION
**Pull request by KEGustafsson**
**Content**
Upgrades to two major components for docker images
- Ubuntu 24.04 LTS base image
- Node upgrade from 20.x LTS to 22.x LTS

I know this is big change and will need lot of further testing.

**Notes:**
Default user `ubuntu:x:1000:` remove and replaced with `node:x:1000:`
Reserved groups in Ubuntu 24.04.  
`
node:x:1000:
systemd-journal:x:999:
systemd-network:x:998:
systemd-timesync:x:997:
input:x:996:
sgx:x:995:
kvm:x:994:
render:x:993:
systemd-resolve:x:992:
`
Therefore old `docker`, `i2c` and `spi` groups need to be changed.

**Testing**
Some `DeprecationWarning` will appear with this upgrade, but that is more or less expected.
So far tested in `amd64` and `arm64` platforms without finding breaking changes.

### CODE REVIEW

**Comment by tkurki** [comment]
I think we should consider this as a whole: upgrade the recommended version, instructions, include v22 (and drop v16) in test runs etc. I am not very eager to upgrade just version in the docker image so that what we recommend and what is in the image are not matching.

But nice to know that things are working!

**Comment by KEGustafsson** [comment]
Agree. Opened this just to prepare things to upgrade. 

Maybe we should consider parallel test image using this instead of changing current setup. At least until more testing and good understanding about impacts.

**Comment by tkurki** [comment]
A separate tagged docker image is a great idea!

**Comment by KEGustafsson** [comment]
Now ready for first review
- base images for Ubuntu 24.04 and Node22.x build automatically once per week
- docker builds on these base images triggered same way as previously
- new SK docker images have `-test` tag added

File: .github/workflows/build-docker-test.yml [hunk 1]
```diff
@@ -0,0 +1,97 @@
+name: Build Docker development test container
```
Comment (tkurki) [inline .github/workflows/build-docker-test.yml]: This is builds an image, right, not a container? So should be named `Build Docker test image`?

**Comment by tkurki** [comment]
To be honest I am a little lost as to what changes and what the practical end result is going to be. Think about it as a regular user running sk on docker: how can they understand what is happening, what’s in it for them and how do we envision to continue from here. The PR will be linked to release notes, so the description plays a role.

And should the docs be updated?

**Comment by KEGustafsson** [comment]
My intetion is/was to bring these new background technologies to wider testing. Target for active developers, not kind of end-users.

Thinking other options and I came to alternative idea. Lets not merge this to master, but e.g. same name branch that I have. Triggering changed to manual and GA could be triggered whenever seen beneficial, not every commit.

Developers could test stuff against this and give feedback.

Then setup would get broader audience and test penetration without messing master.

**Comment by KEGustafsson** [comment]
I'll close this PR and open two new ones.
1) Ubuntu 24.04 upgrade to base image
2) Node22 upgrade

**Pull request closed without merging**